src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -96,14 +95,9 @@ public T remove(int index) {
     @Override
     public T remove(T element) {
         for (int i = 0; i < size; i++) {
-            if (element == null) {
-                if (elementInfo[i] == null) {
-                    return remove(i);
-                }
-            } else {
-                if (element.equals(elementInfo[i])) {
-                    return remove(i);
-                }
+            if (element == null ? elementInfo[i] == null
+                    : element.equals(elementInfo[i])) {
+                return remove(i);
```

Comment:
An interesting solution. But I meant the standard solution, which is in the implementation of Objects.equals() -> `if (element == elementInfo[i] || element != null && element.equals(elementInfo[i])){}`. 
Okay. I will approve this solution.